### DISCUSSION TITLE
useContractWrite not opening metamask popup

### DISCUSSION THREAD

**Discussion by shreyaspapi** [wevm/wagmi#2125] [Q&A]
I am using the useContractWrite hook from the wagmi library to interact with a smart contract on the Ethereum network. However, I am facing an issue where the hook seems to get called even without clicking the button, and the Metamask popup for signing the transaction is not opening.

Here's the code I am using:

```
const { config } = usePrepareContractWrite({
        address: tokenAddress,
        abi: abi,
        functionName: method.name,
        signerOrProvider: provider,
})

const { data, isLoading, isSuccess, write } = useContractWrite(config);
```

And in the UI:

```
<Button variant="contained" disabled={!write} onClick={() => write?.(methodInputs)}>
    Send
</Button>

```

I have created multiple components with the same above code, but useContractWrite seems to get called even without clicking the button. With the following error and button is always disabled.

<img width="536" alt="Screenshot 2023-04-04 at 12 55 20 PM" src="https://user-images.githubusercontent.com/22324802/229719035-d1195e47-307c-4c65-bcae-7ee28f2a231b.png">

I would appreciate any help in resolving this issue. Thank you!

**Comment by victorpavlenko** [comment]
Nobody will help you

**Reply by shreyaspapi** [reply]
😕, the sad part is I went over the code base multiple times not knowing what I did wrong. Had to create a complete clean project to make it work.

**Reply by victorpavlenko** [reply]
@shreyaspapi but u make it work? bit luck

**Reply by SaadNiaziDev** [reply]
> 😕, the sad part is I went over the code base multiple times not knowing what I did wrong. Had to create a complete clean project to make it work.

u got it working or still having problem? Would like to help if your problem is not solved

**Reply by chknlittle** [reply]
i have this issue, no errors, no leads

**Reply by shreyaspapi** [reply]
Yeah I created a new project, and it worked with same code there :)

**Reply by dmulvi** [reply]
Late to the party, but I had this same issue. Figured it out by adding error handling as outlined here: https://wagmi.sh/react/guides/write-to-contract#_7-handle-errors-optional

**Comment by lxiiiixi** [comment]
This problem is because some execution conditions in the contract are not met, which causes the revert function to be triggered, and you should ask the contract developer after sorting out the parameters you passed into this contract function.
For example here in a solidity contract:
<img width="837" alt="image" src="https://github.com/wevm/wagmi/assets/104674636/c3ef13c9-23bc-4f03-914b-785275c6b32f">
if the balance of caller is not enought, contract will revert and trigger some error to to.
